### PR TITLE
Added Strossle to AMP solutions

### PR DESCRIPTION
**Pull request by Serzhenka**
✨Strossle provides technologies and revenue drivers helping publishers monetize the true value of their content. We use machine learning to match the right users with the right content, presented as best in class recommendations. You benefit from more engaged users and multitude of options to grow revenues from advertising and paid content.

Please follow link below: `https://strossle.com`

@rsimha, @torch2424 guys, could you please check and confirm this request?

### CODE REVIEW

**Comment by Serzhenka** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here (e.g. `I signed it!`) and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google.com/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fampproject%2Famphtml%2Fpull%2F22615) for more info**.

I signed it!

File: ads/strossle.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
```
Comment (rsimha) [inline ads/strossle.js]: 2019. Here and everywhere else.

Comment (Serzhenka) [reply ads/strossle.js]: @rsimha Got it thanks! Date was fixed.
@lannka Could you please help us here with review and merging? And sorry for the rush, it's just important case.

File: ads/_config.js [hunk 2]
```diff
@@ -903,6 +903,15 @@ export const adConfig = {
     renderStartImplemented: true,
   },
 
+  'strossle': {
+    preconnect: [
+      'https://amp.spklw.com',
+      'https://widgets.sprinklecontent.com',
+      'https://images.sprinklecontent.com',
+    ],
+    renderStartImplemented: true,
```
Comment (torch2424) [inline ads/_config.js]: So I went ahead and tested your `renderStart` implementation:

Screen Shot 2019-06-03 at 5 07 17 PM

However, my `console.log` inside of the `amp-ad-xorigin-iframe-handler.js` is not called:

Screen Shot 2019-06-03 at 5 08 57 PM

Thus, I just wanted to double check if your server supports `renderStartImplemented`? If possible, could you please implement this? It helps us serve ads as fast as possible to our users, and gives us some big performance gains. If not, could you please remove this line?

Let me know what you think. Thank you! 😄

Comment (Serzhenka) [reply ads/_config.js]: @torch2424 Sure, thanks showing me that line. 
For now we will skip this feature `renderStartImplemented `.

Looks like ok for now after removing that.
Could you please confirm that too?  😄

**Comment by torch2424** [comment]
@Serzhenka Thanks for making the changes for `renderStartImplemented`! 😄 

Now it seems you committed the `yarn.lock` when you made the latest changes, could you please remove the `yarn.lock` changes? As those shouldn't be related to this PR. Thanks! 😄 

After that, I think it will be good to go!

**Comment by Serzhenka** [comment]
> @Serzhenka Thanks for making the changes for `renderStartImplemented`! 😄
> 
> Now it seems you committed the `yarn.lock` when you made the latest changes, could you please remove the `yarn.lock` changes? As those shouldn't be related to this PR. Thanks! 😄
> 
> After that, I think it will be good to go!

@torch2424 Thank you so much!😄And sorry for this error,  fixed now!

**Comment by Serzhenka** [comment]
@torch2424 Could you please come back here? 😄

**Comment by Serzhenka** [comment]
@torch2424 @lannka @rsimha 
Guys, please come back here, and sorry for the rush, we really need this merge!

File: ads/strossle.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * @param {!Window} global
+ * @param {string} data
+ */
+function createContainer(global, data) {
+  const d = global.document.createElement('div');
+  d.className = renderTo;
+  d.setAttribute('data-spklw-widget', data.widgetid);
+  global.document.getElementById('c').appendChild(d);
```
Comment (torch2424) [inline ads/strossle.js]: So just noticed this, can we use a better identifier by any chance for selecting the element?

cc @kristoferbaxter @zhouyx

Comment (Serzhenka) [reply ads/strossle.js]: @torch2424 yarn.lock added back (without any changes from my side), thanks for your comment above. Please confirm that, and if all good for now, let's merge it 😄

Comment (zhouyx) [reply ads/strossle.js]: @torch2424 
This is used to select the container here
https://github.com/ampproject/amphtml/blob/bd9c449ec893b5dd7b51a495c0a69c003394c916/3p/frame.max.html#L8-L10

I don't think there's any identifier except the `id=c`. The selector looks good to me. Thanks.

File: yarn.lock [hunk 4]
```diff
@@ -16168,4 +16168,4 @@ zip-stream@^1.2.0:
     archiver-utils "^1.3.0"
     compress-commons "^1.2.0"
     lodash "^4.8.0"
-    readable-stream "^2.0.0"
+    readable-stream "^2.0.0"
```
Comment (torch2424) [inline yarn.lock]: Hello, so it seems like you copy pasted the `yarn.lock` back into the file, but then removed a newline.

Could you try [checking out the file and reset to the master branch](https://stackoverflow.com/questions/1817766/how-to-revert-to-origins-master-branchs-version-of-file) like: `git checkout origin/master yarn.lock`?

Thank you!

File: ads/strossle.js (hunk 3 above)
Comment (torch2424) [reply ads/strossle.js]: @Serzhenka Thank you! 😄 

@zhouyx Thanks for taking a look! 😄

File: yarn.lock (hunk 4 above)
Comment (Serzhenka) [reply yarn.lock]: @torch2424 Yes, sure!
Hi `yarn.lock` is back from upstream/master as it is.
Previously i've copy paste - yes you was right, sorry for this concern.
Please have a check it again and thanks for you time 😄

Comment (Serzhenka) [reply yarn.lock]: @torch2424 Just a little reminder 😉
Thank you 😊

Comment (Serzhenka) [reply yarn.lock]: @torch2424
Guys, please come back here🙂

**Comment by torch2424** [comment]
@Serzhenka 

Thanks for making the `yarn.lock` fixes! Now it seems like there is one last thing 😢 

https://travis-ci.org/ampproject/amphtml/jobs/542685693#L697

Seems like you have some check-type errors. thus the travis build is failing. You can try running `gulp check-types --local-changes` to reproduce this on your machine.

cc @zhouyx can you keep an eye on this while I am out? 😄

**Comment by zhouyx** [comment]
@Serzhenka Thanks for the Pull Request. Approved and merged.

**Comment by Serzhenka** [comment]
@zhouyx @torch2424 
🙏🏻Thank you so much guys! 😉

**Comment by Serzhenka** [comment]
@zhouyx Could you please tell me how much time do we need to wait till this merge will be available in [https://cdn.ampproject.org/v0/amp-ad-0.1.js](https://cdn.ampproject.org/v0/amp-ad-0.1.js)? May be you have vision based on your experience?

Or if we need to update something else to see Strossle on this CDN - please tell us 😉

**Comment by zhouyx** [comment]
Change in this week's canary cut. Should be out in prod next week. 
In the meanwhile, you can opt-in to our [dev channel](https://cdn.ampproject.org/experiments.html) to test the code. Thanks.

**Comment by rsimha** [comment]
> Seems like you have some check-type errors. thus the travis build is failing. You can try running `gulp check-types --local-changes` to reproduce this on your machine.

@torch2424 There is no `--local-changes` flag for `gulp check-types`. Running it without the flag should work though. Run `gulp help` for a listing of all available flags.

**Comment by torch2424** [comment]
Thank you @rsimha 😄

**Comment by Serzhenka** [comment]
@rsimha Thanks, does it ok for new release today? Or need some update from my side for example?

**Comment by rsimha** [comment]
@Serzhenka This PR is part of https://github.com/ampproject/amphtml/releases/tag/1906111828200, which should be released soon. Until then, you can verify it by opting in to the [RC channel](https://github.com/ampproject/amphtml/blob/master/contributing/release-schedule.md#amp-release-candidate-rc-channel).

**Pull request merged**